### PR TITLE
Stricter docstring styling

### PR DESCRIPTION
**Pull request by provinzkraut**
This enforces stricter linting rules in docstrings. 

Previously, mostly formatting was concerned, this also tackles stylistic choices regarding the language used within docstrings to make it more consistent across the project.

Additionally, an issue with `docformatter` was fixed that needlessly wrapped docstrings to 79 characters, and all docstrings were reformatted to adhere to the new line length. The new line length is now 120, which is consistent with the one used for the rest of the code.

# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [ ] Have you got 100% test coverage on new code?
- [ ] Have you updated the prose documentation?
- [ ] Have you updated the reference documentation?

### CODE REVIEW

File: starlite/datastructures/cookie.py [hunk 1]
```diff
@@ -53,15 +53,14 @@ def to_header(self, **kwargs: Any) -> str:
         return simple_cookie.output(**kwargs).strip()
 
     def __eq__(self, other: Any) -> bool:
-        """Custom implementation to allow using equality operators on class
-        instances.
+        """Determine whether two cookie instances are equal according to the
+        cookie spec, i.e. hey have a similar path, domain and key.
```
Comment (peterschutt) [inline starlite/datastructures/cookie.py]:
```suggestion
        cookie spec, i.e. they have a similar path, domain and key.
```

File: starlite/handlers/http.py [hunk 2]
```diff
@@ -438,13 +441,13 @@ def __init__(
...
     def __call__(self, fn: "AnyCallable") -> "HTTPRouteHandler":
-        """Replaces a function with itself."""
+        """Replace a function with itself."""
         self.fn = fn
         self._validate_handler_function()
         return self
 
     def resolve_response_class(self) -> Type["Response"]:
-        """Returns the closest custom Response class in the owner graph or the
+        """Return the closest custom Response class in the owner graph or the
```
Comment (peterschutt) [inline starlite/handlers/http.py]:
```suggestion
        """Return the closest custom Response class in the ownership graph or the
```

File: starlite/handlers/http.py [hunk 3]
```diff
@@ -758,7 +762,9 @@ def __init__(
             description: Text used for the route's schema description section.
             include_in_schema: A boolean flag dictating whether  the route handler should be documented in the OpenAPI schema.
             operation_id: An identifier used for the route's schema operationId. Defaults to the __name__ of the wrapped function.
-            raises:  A list of exception classes extending from starlite.HttpException that is used for the OpenAPI documentation. This list should describe all exceptions raised within the route handler's function/method. The Starlite ValidationException will be added automatically for the schema if any validation is involved.
+            raises:  A list of exception classes extending from starlite.HttpException that is used for the OpenAPI documentation.
```
Comment (peterschutt) [inline starlite/handlers/http.py:755]: Cross-ref here?

File: starlite/logging/utils.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 
 def resolve_handlers(handlers: List[Any]) -> List[Any]:
-    """Converts list of string of handlers to the object of respective handler.
+    """Convert list of string of handlers to the object of respective handler.
```
Comment (peterschutt) [inline starlite/logging/utils.py:5]:
```suggestion
    """Convert list of string handler names to the object of respective handler.
```
Not sure if my suggestion is correct, but this reads awkward.

File: starlite/middleware/compression.py [hunk 5]
```diff
@@ -151,7 +151,7 @@ def create_compression_send_wrapper(
         send: "Send",
         compression_encoding: "Literal[CompressionEncoding.BROTLI, CompressionEncoding.GZIP]",
     ) -> "Send":
-        """Wraps 'send' to handle brotli compression.
+        """Wrap 'send' to handle brotli compression.
```
Comment (peterschutt) [inline starlite/middleware/compression.py:151]:
```suggestion
        """Wrap `send` to handle brotli compression.
```

File: starlite/middleware/compression.py [hunk 6]
```diff
@@ -168,7 +168,7 @@ def create_compression_send_wrapper(
         started = Ref[bool](False)
 
         async def send_wrapper(message: "Message") -> None:
-            """Handles and compresses the HTTP Message with brotli.
+            """Handle and compresses the HTTP Message with brotli.
```
Comment (peterschutt) [inline starlite/middleware/compression.py:168]:
```suggestion
            """Handle and compress the HTTP Message with brotli.
```

File: starlite/middleware/exceptions/middleware.py [hunk 7]
```diff
@@ -35,7 +34,7 @@ def __init__(self, app: "ASGIApp", debug: bool, exception_handlers: "ExceptionHa
         self.debug = debug
 
     async def __call__(self, scope: "Scope", receive: "Receive", send: "Send") -> None:
-        """The middleware's ASGI-callable.
+        """ASGI-callable.
```
Comment (peterschutt) [inline starlite/middleware/exceptions/middleware.py:36]:
```suggestion
        """ASGI callable.
```
No hyphen used in previous similar doc strings.

File: starlite/middleware/session/base.py [hunk 8]
```diff
@@ -391,7 +392,7 @@ async def wrapped_send(message: "Message") -> None:
         return wrapped_send
 
     async def __call__(self, scope: "Scope", receive: "Receive", send: "Send") -> None:
-        """The middleware's ASGI-callable.
+        """ASGI-callable.
```
Comment (peterschutt) [inline starlite/middleware/session/base.py:389]:
```suggestion
        """ASGI callable.
```

File: starlite/openapi/path_item.py [hunk 9]
```diff
@@ -20,7 +20,10 @@
 
 
 def get_description_for_handler(route_handler: "HTTPRouteHandler", use_handler_docstrings: bool) -> Optional[str]:
-    """Produces the operation description for a route handler, either by using the description value if provided, or the docstring - if config is enabled.
+    """Produce the operation description for a route handler, either by using
+    the description value if provided,
+
+    or the docstring - if config is enabled.
```
Comment (peterschutt) [inline starlite/openapi/path_item.py:25]: line break?

**Review by Goldziher — APPROVED** [review]

File: starlite/plugins/sql_alchemy/plugin.py [hunk 10]
```diff
@@ -83,7 +84,7 @@ def on_app_init(self, app: "Starlite") -> None:
 
     @staticmethod
     def is_plugin_supported_type(value: Any) -> "TypeGuard[DeclarativeMeta]":
-        """A typeguard that tests whether values are subclasses of SQLAlchemy's
+        """`TypeGuard` testing whether values are subclasses of SQLAlchemy's
         'DeclarativeMeta' class.
```
Comment (peterschutt) [inline starlite/plugins/sql_alchemy/plugin.py]:
```suggestion
        `DeclarativeMeta` class.
```

File: starlite/response/base.py [hunk 11]
```diff
@@ -109,7 +109,7 @@ def set_cookie(
         httponly: bool = False,
         samesite: 'Literal["lax", "strict", "none"]' = "lax",
     ) -> None:
-        """Sets a cookie on the response.
+        """Ses a cookie on the response.
```
Comment (peterschutt) [inline starlite/response/base.py:110]:
```suggestion
        """Sets a cookie on the response.
```

File: starlite/response/file.py [hunk 12]
```diff
@@ -97,7 +97,7 @@ def __init__(
         stat_result: Optional["stat_result_type"] = None,
         status_code: int = HTTP_200_OK,
     ) -> None:
-        """This class allows streaming a file as response body.
+        """Initialize `FileResponse`
```
Comment (peterschutt) [inline starlite/response/file.py:100]:
```suggestion
        """Initialize `FileResponse`.
```

File: starlite/routes/http.py [hunk 13]
```diff
@@ -168,8 +172,8 @@ async def _call_handler_function(
     async def _get_response_data(
         route_handler: "HTTPRouteHandler", parameter_model: "KwargsModel", request: Request
     ) -> Any:
-        """Determines what kwargs are required for the given route handler's
-        'fn' and calls it.
+        """Determine what kwargs are required for the given route handler's
+        `fn` and calls it.
```
Comment (peterschutt) [inline starlite/routes/http.py]:
```suggestion
        `fn` and call it.
```

File: starlite/utils/dependency.py [hunk 14]
```diff
@@ -42,9 +42,7 @@ def should_skip_dependency_validation(val: Any) -> bool:
 async def _resolve_dependency_into_kwargs(
     model: "KwargsModel", dependency: "Dependency", connection: Union["WebSocket", "Request"], kwargs: Dict[str, Any]
 ) -> None:
-    """Helper function to store the result of resolve_dependency in the
-    kwargs.
-    """
+    """Store the result of resolve_dependency in the kwargs."""
```
Comment (peterschutt) [inline starlite/utils/dependency.py:44]:
```suggestion
    """Store the result of `resolve_dependency` in the `kwargs`."""
```

**Review by peterschutt — APPROVED** [review]

File: starlite/response/file.py (hunk 12 above)
Comment (provinzkraut) [reply starlite/response/file.py:100]: This is weird, that should have been caught by `pydocstyle`. Wonder why it ignores that.

File: starlite/response/base.py (hunk 11 above)
Comment (peterschutt) [reply starlite/response/base.py:110]: Actually should be `Set` to be consistent.

**Pull request merged**